gyb_syntax_support/Node.py

```diff
@@ -123,3 +123,57 @@ def is_token(self):
         """
         return 'Token' in self.syntax_kind or \
             'Token' in self.collection_element
+
+    def has_optional_base_type_child(self):
+      return any(child.has_optional_base_type() for child in self.children)
+
+    def generate_initializer_decl(self, include_optional_base, current_indentation=""):
+      if not self.children:
+        return "public init()"
+
+      generic_param_names = {}
+      generic_params = []
+      params = []
+
+      skip_next = False
+      if self.children:
+        params.append("leadingTrivia: Trivia? = nil")
+      for child in self.children:
+        if skip_next:
+          skip_next = False
+          continue
+
+        if child.has_optional_base_type() and not include_optional_base:
+          skip_next = True
```

Comment:
This is to skip the unexpected nodes, right? Could you add a comment for that?

Reply:
Yep, will do.